src/shogun/machine/gp/FITCInferenceMethod.cpp

```diff
@@ -76,17 +74,26 @@ void CFITCInferenceMethod::update()
 	SG_DEBUG("leaving\n");
 }
 
+CFITCInferenceMethod* CFITCInferenceMethod::obtain_from_generic(
+		CInferenceMethod* inference)
+{
+	ASSERT(inference!=NULL);
+
+	if (inference->get_inference_type()!=INF_FITC_REGRESSION)
+		SG_SERROR("Provided inference is not of type CFITCInferenceMethod!\n")
+
+	SG_REF(inference);
+	return (CFITCInferenceMethod*)inference;
+}
+
 void CFITCInferenceMethod::check_members() const
 {
-	CInferenceMethod::check_members();
+	CSingleFITCLaplacianBase::check_members();
 
 	REQUIRE(m_model->get_model_type()==LT_GAUSSIAN,
 			"FITC inference method can only use Gaussian likelihood function\n")
 	REQUIRE(m_labels->get_label_type()==LT_REGRESSION, "Labels must be type "
 			"of CRegressionLabels\n")
-	REQUIRE(m_inducing_features, "inducing features should not be NULL\n")
```

Comment:
The reason is the base class has to the same checking job since `m_inducing_features` is a protected member of the base class.